.github/workflows/regression.yml

```diff
@@ -149,33 +293,478 @@ jobs:
+        id: read-analysis
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ${{ runner.temp }}/outputs/report.html
+
+      - name: Post report to PR
+        uses: peter-evans/create-or-update-comment@v3
+        with:
+          issue-number: ${{ needs.compute-metadata.outputs.pr-number }}
+          edit-mode: append
+          body: ${{ steps.read-analysis.outputs.content }}
+
+      - name: Upload regression report to artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: meta
-          path: "${{ runner.temp }}/meta"
+          name: capture-artifacts
+          path: ${{ runner.temp }}/outputs/*
+
+      - name: Check status, cancelled
+        if: ${{ cancelled() }}
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            /repos/${{ github.repository }}/statuses/${{ needs.compute-metadata.outputs.comparison-sha }} \
+            -f state='failure' \
+            -f description='Analyze experimental results from Regression Detection Suite cancelled.' \
+            -f context='Regression Detection Suite / analyze-experiment' \
+            -f target_url=${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+      - name: Check status, success
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            /repos/${{ github.repository }}/statuses/${{ needs.compute-metadata.outputs.comparison-sha }} \
+            -f state='success' \
+            -f description='Analyze experimental results from Regression Detection Suite succeeded.' \
+            -f context='Regression Detection Suite / analyze-experiment' \
+            -f target_url=${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+      - name: Check status, failure
+        if: ${{ failure() }}
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            /repos/${{ github.repository }}/statuses/${{ needs.compute-metadata.outputs.comparison-sha }} \
+            -f state='failure' \
+            -f description='Analyze experimental results from Regression Detection Suite failed.' \
+            -f context='Regression Detection Suite / analyze-experiment' \
+            -f target_url=${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+  regression-detection-suite:
+    name: Regression Detection Suite
+    runs-on: ubuntu-latest
+    if: |
+      always() &&
+        (github.event_name != 'issue_comment' || (github.event.issue.pull_request && contains(github.event.comment.body, '/ci-run-regression')))
```

Comment:
I actually chose `!=' intentionally here because the only time we don't want to always run this job (🙄) is if there is an issue_comment made, which doesn't match the required text.
i.e if we added a cron schedule trigger to this workflow, this logic would still be sound.

And if we flipped the logic to be '==' , then we'd need to specify the non-issue_comment triggers here explicitly (thereby being prone to being forgotten to be updated at a later time).

I really wish the `issue_comment` trigger had a setting before the jobs, to filter on body text, that would avoid having to have this nonsense, and it could just be "always()".

Reply:
Ah, gotcha. Would the refactoring to have a workflow that _only_ looks for issue comments, which would then trigger this workflow, help here? It seems like it would simplify things.

Reply:
Good call. Yes that should help. I'll test that.

Reply:
Implemented this.